compiler/rustc_trait_selection/src/traits/wf.rs

```diff
@@ -430,7 +430,7 @@ impl<'a, 'tcx> WfPredicates<'a, 'tcx> {
 
                 GenericArgKind::Const(constant) => {
                     match constant.val {
-                        ty::ConstKind::Unevaluated(def, substs, promoted) => {
+                        ty::ConstKind::Unevaluated(ty::Unevaluated { def, substs, promoted }) => {
```

Comment:
PredicateKind::ConstEvaluatable could take a `ty::Unevaluated`, but considering that it can't take promoteds... maybe not

Reply:
once `val-tree` is further ahead we should do that :+1: